runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -209,6 +209,7 @@ public void clearAll() {
      * and make it look like the table is a newly opened one
      * This is useful for JVMs who do not wish to keep the entire table
      * around in memory.
+     * DO NOT USE THIS METHOD WITH NO_CACHE TABLES
```

Comment:
Instead of relying on this comment, can we throw an IllegalStateException if this is the case? It will be easier to detect incorrect usage patterns.

Reply:
As you have mentioned offline, there could be a future case where a table is opened with CACHE and NO_CACHE separately within the same runtime. I think this would make for a good test to validate that this method behaves as expected for both opens.

Reply:
I have added the check in the beginning of this method. I can't write a test for this check because today there is no way to open a Table using NO_CACHE mode.